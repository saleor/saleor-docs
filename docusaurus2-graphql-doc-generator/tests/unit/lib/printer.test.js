const { capitalize } = require("lodash");

const Printer = require("@/lib/printer");

jest.mock("@/lib/graphql");
const graphql = require("@/lib/graphql");

const graphQLEntityType = [
  "enum",
  "union",
  "interface",
  "object",
  "input",
  "scalar",
  "directive",
];

describe("lib", () => {
  describe("printer", () => {
    describe("class Printer", () => {
      let printerInstance,
        baseURL = "graphql",
        root = "docs",
        schema = { toString: () => "SCHEMA", getType: (type) => type };

      beforeEach(() => {
        printerInstance = new Printer(schema, baseURL, root);
      });

      afterEach(() => {
        jest.resetAllMocks();
      });

      describe("toLink()", () => {
        test.each(graphQLEntityType)(
          "returns markdown link for GraphQL %s",
          (type) => {
            const entityName = `Test${capitalize(type)}`;
            jest
              .spyOn(graphql, `is${capitalize(type)}Type`)
              .mockReturnValueOnce(true);
            jest.spyOn(graphql, "getNamedType").mockReturnValue(entityName);
            const link = printerInstance.toLink(type, entityName);
            expect(link).toMatchSnapshot();
          },
        );

        test("returns markdown link surrounded by [] for GraphQL list/array", () => {
          const type = "list";
          const subtype = "object";
          const entityName = "TestObjectList";
          jest.spyOn(graphql, "isListType").mockReturnValue(true);
          jest
            .spyOn(graphql, `is${capitalize(subtype)}Type`)
            .mockReturnValueOnce(true);
          jest.spyOn(graphql, "getNamedType").mockReturnValue(entityName);
          const link = printerInstance.toLink(type, entityName);
          expect(link).toMatchSnapshot();
        });

        test("returns plain text for unknown entities", () => {
          const type = "any";
          const entityName = "fooBar";
          const link = printerInstance.toLink(type, entityName);
          expect(link).toMatchSnapshot();
        });
      });

      describe("printSection()", () => {
        test("returns Markdown ### section by default", () => {
          const printSectionItems = jest
            .spyOn(printerInstance, "printSectionItems")
            .mockImplementation((content) => content);
          const title = "section title";
          const content = "section content";
          const section = printerInstance.printSection(content, title);
          expect(printSectionItems).toHaveBeenCalledWith(content);
          expect(section).toMatchSnapshot();
        });

        test("returns Markdown custom section level", () => {
          jest
            .spyOn(printerInstance, "printSectionItems")
            .mockImplementation((content) => content);
          const title = "section title";
          const content = "section content";
          const section = printerInstance.printSection(content, title, "#");
          expect(section).toMatchSnapshot();
        });

        test("returns empty string if content is empty", () => {
          const title = "section title";
          const content = "";
          const section = printerInstance.printSection(content, title);
          expect(section).toBe("");
        });
      });

      describe("printSectionItems()", () => {
        test("returns Markdown one line per item", () => {
          const printSectionItem = jest
            .spyOn(printerInstance, "printSectionItem")
            .mockImplementation((content) => content);

          const itemList = ["one", "two", "three"];
          const section = printerInstance.printSectionItems(itemList);
          expect(printSectionItem).toHaveBeenCalledTimes(3);
          expect(printSectionItem).toHaveBeenLastCalledWith(
            itemList.pop(),
            "####",
          );
          expect(section).toMatchSnapshot();
        });

        test("returns empty text if not a list", () => {
          const itemList = "list";
          const section = printerInstance.printSectionItems(itemList);
          expect(section).toMatch("");
        });
      });

      describe("printSectionItem()", () => {
        test("returns Markdown #### link section with description", () => {
          const type = { name: "EntityTypeName", description: "Lorem ipsum" };
          jest.spyOn(graphql, "getTypeName").mockReturnValue(type.name);
          jest.spyOn(graphql, "getNamedType").mockReturnValue(type.name);
          jest.spyOn(graphql, "isObjectType").mockReturnValueOnce(true);

          const section = printerInstance.printSectionItem(type);
          expect(section).toMatchSnapshot();
        });

        test("returns Markdown #### link section with sub type", () => {
          const type = {
            name: "EntityTypeName",
            type: { name: "AnObjectType" },
          };
          jest
            .spyOn(graphql, "getTypeName")
            .mockImplementation((type) => type.name);
          jest
            .spyOn(graphql, "getNamedType")
            .mockImplementation((type) => type.name);
          jest
            .spyOn(graphql, "isObjectType")
            .mockReturnValueOnce(false)
            .mockReturnValueOnce(true);

          const section = printerInstance.printSectionItem(type);
          expect(section).toMatchSnapshot();
        });

        test("returns Markdown #### link section with field parameters", () => {
          const type = {
            name: "EntityTypeName",
            args: [{ name: "ParameterTypeName" }],
          };
          jest
            .spyOn(graphql, "getTypeName")
            .mockImplementation((type) => type.name);
          jest
            .spyOn(graphql, "getNamedType")
            .mockImplementation((type) => type.name);
          jest.spyOn(graphql, "isObjectType").mockReturnValueOnce(true);
          jest.spyOn(graphql, "isParametrizedField").mockReturnValueOnce(true);
          const printSectionItems = jest.spyOn(
            printerInstance,
            "printSectionItems",
          );

          const section = printerInstance.printSectionItem(type);
          expect(printSectionItems).toHaveBeenCalledWith(type.args, "- #####");
          expect(section).toMatchSnapshot();
        });
      });

      describe("printCodeEnum()", () => {
        const type = {
          name: "EnumTypeName",
          getValues: () => [{ name: "one" }, { name: "two" }],
        };

        test("returns enum code structure", () => {
          jest.spyOn(graphql, "isEnumType").mockReturnValueOnce(true);
          jest
            .spyOn(graphql, "getTypeName")
            .mockImplementation((type) => type.name);
          const code = printerInstance.printCodeEnum(type);
          expect(code).toMatchSnapshot();
        });

        test("returns empty string if not enum type", () => {
          jest.spyOn(graphql, "isEnumType").mockReturnValueOnce(false);
          const code = printerInstance.printCodeEnum(type);
          expect(code).toBe("");
        });
      });

      describe("printCodeUnion()", () => {
        const type = {
          name: "UnionTypeName",
          getTypes: () => [{ name: "one" }, { name: "two" }],
        };

        test("returns union code structure", () => {
          jest.spyOn(graphql, "isUnionType").mockReturnValueOnce(true);
          jest
            .spyOn(graphql, "getTypeName")
            .mockImplementation((type) => type.name);
          const code = printerInstance.printCodeUnion(type);
          expect(code).toMatchSnapshot();
        });

        test("returns empty string if not union type", () => {
          jest.spyOn(graphql, "isUnionType").mockReturnValueOnce(false);
          const code = printerInstance.printCodeUnion(type);
          expect(code).toBe("");
        });
      });

      describe("printCodeScalar()", () => {
        const type = {
          name: "ScalarTypeName",
        };

        test("returns scalar code structure", () => {
          jest
            .spyOn(graphql, "getTypeName")
            .mockImplementation((type) => type.name);
          const code = printerInstance.printCodeScalar(type);
          expect(code).toMatchSnapshot();
        });
      });

      describe("printCodeArguments()", () => {
        test("returns a Markdown one line per formatted argument with default value surrounded by ()", () => {
          const type = {
            name: "OperationName",
            args: [
              {
                name: "ParamWithDefault",
                type: "string",
                default: "defaultValue",
              },
              { name: "ParamNoDefault", type: "any" },
            ],
          };
          jest
            .spyOn(graphql, "getDefaultValue")
            .mockImplementation((type) => type.default || null);
          const code = printerInstance.printCodeArguments(type);
          expect(code).toMatchSnapshot();
        });

        test("returns an empty string if args list empty", () => {
          const type = {
            name: "OperationName",
            args: [],
          };
          const code = printerInstance.printCodeArguments(type);
          expect(code).toBe("");
        });

        test("returns an empty string if no args", () => {
          const type = {
            name: "OperationName",
          };
          const code = printerInstance.printCodeArguments(type);
          expect(code).toBe("");
        });
      });

      describe("printCodeField()", () => {
        test("returns a field with its type", () => {
          const type = { name: "FooBar", type: "string" };
          jest
            .spyOn(graphql, "getTypeName")
            .mockReturnValueOnce(type.name)
            .mockReturnValueOnce(type.type);
          jest
            .spyOn(printerInstance, "printCodeArguments")
            .mockReturnValueOnce("");
          const code = printerInstance.printCodeField(type);
          expect(code).toMatchSnapshot();
        });

        test("returns a field with its type and arguments", () => {
          const type = { name: "FooBar", type: "string" };
          jest
            .spyOn(graphql, "getTypeName")
            .mockReturnValueOnce(type.name)
            .mockReturnValueOnce(type.type);
          jest
            .spyOn(printerInstance, "printCodeArguments")
            .mockReturnValueOnce("(\narg: boolean\n)");
          const code = printerInstance.printCodeField(type);
          expect(code).toMatchSnapshot();
        });
      });

      describe("printCodeDirective()", () => {
        test("returns a directive", () => {
          const type = { name: "FooBar", type: "string" };
          jest.spyOn(graphql, "getTypeName").mockReturnValueOnce(type.name);
          jest
            .spyOn(printerInstance, "printCodeArguments")
            .mockReturnValueOnce("");
          const code = printerInstance.printCodeDirective(type);
          expect(code).toMatchSnapshot();
        });

        test("returns a directive with its arguments", () => {
          const type = { name: "FooBar", type: "string" };
          jest.spyOn(graphql, "getTypeName").mockReturnValueOnce(type.name);
          jest
            .spyOn(printerInstance, "printCodeArguments")
            .mockReturnValueOnce("(\narg: boolean\n)");
          const code = printerInstance.printCodeDirective(type);
          expect(code).toMatchSnapshot();
        });
      });

      describe("printCodeType()", () => {
        test("returns an interface with its fields", () => {
          const type = { name: "TestInterfaceName", fields: ["one", "two"] };
          jest.spyOn(graphql, "isInterfaceType").mockReturnValueOnce(true);
          jest.spyOn(graphql, "getTypeName").mockReturnValueOnce()(type.name);
          jest.spyOn(graphql, "getFields").mockReturnValueOnce(type.fields);
          jest
            .spyOn(printerInstance, "printCodeField")
            .mockImplementation((name) => `${name}\n`);
          const code = printerInstance.printCodeType(type);
          expect(code).toMatchSnapshot();
        });

        test("returns an object with its fields and interfaces", () => {
          const type = {
            name: "TestName",
            fields: ["one", "two"],
            getInterfaces: () => [{ name: "TestInterfaceName" }],
          };
          jest.spyOn(graphql, "isInterfaceType").mockReturnValueOnce(false);
          jest
            .spyOn(graphql, "getTypeName")
            .mockImplementation((type) => type.name);
          jest.spyOn(graphql, "getFields").mockReturnValueOnce(type.fields);
          jest
            .spyOn(printerInstance, "printCodeField")
            .mockImplementation((name) => `${name}\n`);
          const code = printerInstance.printCodeType(type);
          expect(code).toMatchSnapshot();
        });
      });

      describe("printHeader()", () => {
        test("returns a Docusaurus document header", () => {
          const header = printerInstance.printHeader(
            "an-object-type-name",
            "An Object Type Name",
          );
          expect(header).toMatchSnapshot();
        });
      });

      describe("printDescription()", () => {
        test("returns the type description text", () => {
          const type = { description: "Lorem ipsum" };
          const description = printerInstance.printDescription(type);
          expect(description).toMatchSnapshot();
        });

        test("returns the default text if no description", () => {
          const type = {};
          const description = printerInstance.printDescription(type);
          expect(description).toMatchSnapshot();
        });

        test("return DEPRECATED tag is deprecated", () => {
          const type = {
            description: "Lorem ipsum",
            isDeprecated: true,
            deprecationReason: "Foobar",
          };
          const description = printerInstance.printDescription(type);
          expect(description).toMatchSnapshot();
        });
      });

      describe("printCode()", () => {
        test.each(graphQLEntityType)(
          "returns a Markdown graphql codeblock with type %s",
          (type) => {
            const entityName = `Test${capitalize(type)}`;
            jest
              .spyOn(graphql, `is${capitalize(type)}Type`)
              .mockReturnValueOnce(true);
            let printCode;
            if (["interface", "object", "input"].includes(type)) {
              printCode = jest
                .spyOn(printerInstance, "printCodeType")
                .mockReturnValueOnce(entityName);
            } else {
              printCode = jest
                .spyOn(printerInstance, `printCode${capitalize(type)}`)
                .mockReturnValueOnce(entityName);
            }
            const code = printerInstance.printCode(entityName);
            expect(printCode).toHaveBeenCalledWith(entityName);
            expect(code).toMatchSnapshot();
          },
        );

        test("returns a Markdown graphql codeblock with type query", () => {
          const entityName = "TestQuery";
          jest.spyOn(graphql, "isOperation").mockReturnValueOnce(true);
          const printCode = jest
            .spyOn(printerInstance, "printCodeField")
            .mockReturnValueOnce(entityName);
          const code = printerInstance.printCode(entityName);
          expect(printCode).toHaveBeenCalledWith(entityName);
          expect(code).toMatchSnapshot();
        });

        test("returns a Markdown codeblock with non supported message for unsupported type", () => {
          const entityName = "TestFooBarType";
          jest.spyOn(graphql, "getTypeName").mockReturnValueOnce(entityName);
          const code = printerInstance.printCode(entityName);
          expect(code).toMatchSnapshot();
        });
      });

      describe("printType()", () => {
        test.each(graphQLEntityType)(
          "returns a Markdown formatted Docusaurus content for type %s",
          (type) => {
            const entityType = {
              name: type,
              getValues: () => {},
              getTypes: () => {},
            };
            jest
              .spyOn(graphql, `is${capitalize(type)}Type`)
              .mockReturnValueOnce(true);
            jest
              .spyOn(printerInstance, "printHeader")
              .mockImplementation((name) => `header-${name}`);
            jest
              .spyOn(printerInstance, "printDescription")
              .mockImplementation((type) => `Test ${capitalize(type.name)}`);
            jest
              .spyOn(printerInstance, "printCode")
              .mockImplementation((type) => `\`\`\`${type.name}\`\`\``);
            jest
              .spyOn(printerInstance, "printSection")
              .mockImplementation((_, section) => section);
            const printedType = printerInstance.printType(
              entityType.name,
              entityType,
            );
            expect(printedType).toMatchSnapshot();
          },
        );

        test.each(["object", "input"])(
          "returns a Markdown formatted Docusaurus content for %s implementing interface",
          (type) => {
            const entityType = {
              name: type,
              getValues: () => {},
              getTypes: () => {},
              getInterfaces: () => {},
            };
            jest
              .spyOn(graphql, `is${capitalize(type)}Type`)
              .mockReturnValueOnce(true);
            jest
              .spyOn(printerInstance, "printHeader")
              .mockImplementation((name) => `header-${name}\n\n`);
            jest
              .spyOn(printerInstance, "printDescription")
              .mockImplementation(
                (type) => `Test ${capitalize(type.name)}\n\n`,
              );
            jest
              .spyOn(printerInstance, "printCode")
              .mockImplementation((type) => `\`\`\`${type.name}\`\`\`\n\n`);
            jest
              .spyOn(printerInstance, "printSection")
              .mockImplementation((_, section) => `${section}\n\n`);
            const printedType = printerInstance.printType(
              entityType.name,
              entityType,
            );
            expect(printedType).toMatchSnapshot();
          },
        );

        test("returns a Markdown formatted Docusaurus content for query", () => {
          const entityType = {
            name: "query",
          };
          jest.spyOn(graphql, "isOperation").mockReturnValue(true);
          jest.spyOn(graphql, "getTypeName").mockReturnValue(entityType.name);
          jest
            .spyOn(printerInstance, "printHeader")
            .mockImplementation((name) => `header-${name}\n\n`);
          jest
            .spyOn(printerInstance, "printDescription")
            .mockImplementation((type) => `Test ${capitalize(type.name)}\n\n`);
          jest
            .spyOn(printerInstance, "printCode")
            .mockImplementation((type) => `\`\`\`${type.name}\`\`\`\n\n`);
          jest
            .spyOn(printerInstance, "printSection")
            .mockImplementation((_, section) => `${section}\n\n`);
          const printedType = printerInstance.printType(
            entityType.name,
            entityType,
          );
          expect(printedType).toMatchSnapshot();
        });

        test("returns an empty string if no type", () => {
          const printedType = printerInstance.printType("any", null);
          expect(printedType).toBe("");
        });
      });
    });
  });
});
