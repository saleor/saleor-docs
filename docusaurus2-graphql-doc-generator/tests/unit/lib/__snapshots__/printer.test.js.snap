// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib printer class Printer printCode() returns a Markdown codeblock with non supported message for unsupported type 1`] = `
"
\`\`\`graphql
\\"TestFooBarType\\" not supported
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type directive 1`] = `
"
\`\`\`graphql
TestDirective
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type enum 1`] = `
"
\`\`\`graphql
TestEnum
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type input 1`] = `
"
\`\`\`graphql
TestInput
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type interface 1`] = `
"
\`\`\`graphql
TestInterface
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type object 1`] = `
"
\`\`\`graphql
TestObject
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type query 1`] = `
"
\`\`\`graphql
TestQuery
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type scalar 1`] = `
"
\`\`\`graphql
TestScalar
\`\`\`
"
`;

exports[`lib printer class Printer printCode() returns a Markdown graphql codeblock with type union 1`] = `
"
\`\`\`graphql
TestUnion
\`\`\`
"
`;

exports[`lib printer class Printer printCodeArguments() returns a Markdown one line per formatted argument with default value surrounded by () 1`] = `
"(
  ParamWithDefault: string = defaultValue
  ParamNoDefault: any
)"
`;

exports[`lib printer class Printer printCodeDirective() returns a directive 1`] = `"directive @FooBar"`;

exports[`lib printer class Printer printCodeDirective() returns a directive with its arguments 1`] = `
"directive @FooBar(
arg: boolean
)"
`;

exports[`lib printer class Printer printCodeEnum() returns enum code structure 1`] = `
"enum EnumTypeName {
  one
  two
}"
`;

exports[`lib printer class Printer printCodeField() returns a field with its type 1`] = `
"FooBar: string
"
`;

exports[`lib printer class Printer printCodeField() returns a field with its type and arguments 1`] = `
"FooBar(
arg: boolean
): string
"
`;

exports[`lib printer class Printer printCodeScalar() returns scalar code structure 1`] = `"scalar ScalarTypeName"`;

exports[`lib printer class Printer printCodeType() returns an interface with its fields 1`] = `
"interface undefined {
  one
  two
}"
`;

exports[`lib printer class Printer printCodeType() returns an object with its fields and interfaces 1`] = `
"type TestName implements TestInterfaceName {
  one
  two
}"
`;

exports[`lib printer class Printer printCodeUnion() returns union code structure 1`] = `"union UnionTypeName = one | two"`;

exports[`lib printer class Printer printDescription() return DEPRECATED tag is deprecated 1`] = `
"<sub><sup><Tag color=\\"#ffba00\\">DEPRECATED</Tag> Foobar</sup></sub>

Lorem ipsum"
`;

exports[`lib printer class Printer printDescription() returns the default text if no description 1`] = `"No description"`;

exports[`lib printer class Printer printDescription() returns the type description text 1`] = `"Lorem ipsum"`;

exports[`lib printer class Printer printHeader() returns a Docusaurus document header 1`] = `
"---
id: an-object-type-name
title: An Object Type Name
---
"
`;

exports[`lib printer class Printer printSection() returns Markdown ### section by default 1`] = `
"### section title

section content

"
`;

exports[`lib printer class Printer printSection() returns Markdown custom section level 1`] = `
"# section title

section content

"
`;

exports[`lib printer class Printer printSectionItem() returns Markdown #### link section with description 1`] = `
"#### [\`EntityTypeName\`](docs/graphql/objects/entity-type-name) 

Lorem ipsum
"
`;

exports[`lib printer class Printer printSectionItem() returns Markdown #### link section with field parameters 1`] = `
"#### [\`EntityTypeName\`](docs/graphql/objects/entity-type-name) 


- ##### \`ParameterTypeName\` 


"
`;

exports[`lib printer class Printer printSectionItem() returns Markdown #### link section with sub type 1`] = `
"#### \`EntityTypeName\` ([\`AnObjectType\`](docs/graphql/objects/an-object-type))


"
`;

exports[`lib printer class Printer printSectionItems() returns Markdown one line per item 1`] = `
"one

two

three"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for input implementing interface 1`] = `
"header-input

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Input

\`input\`

Fields

Interfaces
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for object implementing interface 1`] = `
"header-object

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Object

\`object\`

Fields

Interfaces
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for query 1`] = `
"header-query

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Query

\`query\`

Arguments

Type
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for type directive 1`] = `
"header-directive

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Directive

\`directive\`

Arguments
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for type enum 1`] = `
"header-enum

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Enum

\`enum\`

Values
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for type input 1`] = `
"header-input

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Input

\`input\`

Fields
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for type interface 1`] = `
"header-interface

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Interface

\`interface\`

Fields
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for type object 1`] = `
"header-object

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Object

\`object\`

Fields
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for type scalar 1`] = `
"header-scalar

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Scalar

\`scalar\`
"
`;

exports[`lib printer class Printer printType() returns a Markdown formatted Docusaurus content for type union 1`] = `
"header-union

export const Tag = ({children, color}) => (
<span
style={{
      backgroundColor: color,
      borderRadius: '2px',
      color: '#fff',
      padding: '0.2rem',
    }}>
{children}
</span>
);

Test Union

\`union\`

Possible types
"
`;

exports[`lib printer class Printer toLink() returns markdown link for GraphQL directive 1`] = `"[\`TestDirective\`](docs/graphql/directives/test-directive)"`;

exports[`lib printer class Printer toLink() returns markdown link for GraphQL enum 1`] = `"[\`TestEnum\`](docs/graphql/enums/test-enum)"`;

exports[`lib printer class Printer toLink() returns markdown link for GraphQL input 1`] = `"[\`TestInput\`](docs/graphql/inputs/test-input)"`;

exports[`lib printer class Printer toLink() returns markdown link for GraphQL interface 1`] = `"[\`TestInterface\`](docs/graphql/interfaces/test-interface)"`;

exports[`lib printer class Printer toLink() returns markdown link for GraphQL object 1`] = `"[\`TestObject\`](docs/graphql/objects/test-object)"`;

exports[`lib printer class Printer toLink() returns markdown link for GraphQL scalar 1`] = `"[\`TestScalar\`](docs/graphql/scalars/test-scalar)"`;

exports[`lib printer class Printer toLink() returns markdown link for GraphQL union 1`] = `"[\`TestUnion\`](docs/graphql/unions/test-union)"`;

exports[`lib printer class Printer toLink() returns markdown link surrounded by [] for GraphQL list/array 1`] = `"[\`TestObjectList\`](docs/graphql/objects/test-object-list)"`;

exports[`lib printer class Printer toLink() returns plain text for unknown entities 1`] = `"\`fooBar\`"`;
